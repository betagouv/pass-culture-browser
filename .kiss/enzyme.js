import React from 'react'
import { shallow } from 'enzyme'

import MyComponent from '../MyComponent'

describe('src | components | MyComponent', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      const props = { errors: {} }
      const wrapper = shallow(<MyComponent {...props} />)
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
