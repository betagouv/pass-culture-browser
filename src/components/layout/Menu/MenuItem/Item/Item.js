import PropTypes from 'prop-types'
import React, { Fragment } from 'react'

import Icon from '../../../Icon'

const Item = ({ icon, title }) => (
  <Fragment>
    <span className="flex-0 text-center menu-icon mr16">
      <Icon
        alt=""
        svg={`ico-${icon}`}
      />
    </span>
    <span className="flex-1 is-medium">{title}</span>
  </Fragment>
)

Item.propTypes = {
  icon: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
}

export default Item
