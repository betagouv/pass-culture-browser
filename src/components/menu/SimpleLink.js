import PropTypes from 'prop-types'
import React from 'react'

import Item from './Item'

const SimpleLink = ({ item }) => (
  <a
    className="navlink mx12 flex-columns"
    disabled={item.disabledInMenu}
    href={item.href}
    key={item.href}
    rel={(item.target === '_blank' && 'noopener noreferer') || null}
    target={item.target}
  >
    <Item
      icon={item.icon}
      title={item.title}
    />
  </a>
)

SimpleLink.propTypes = {
  item: PropTypes.shape({
    disabled: PropTypes.bool,
    href: PropTypes.string.isRequired,
    icon: PropTypes.string.isRequired,
    target: PropTypes.string,
    title: PropTypes.string.isRequired,
  }).isRequired,
}

export default SimpleLink
