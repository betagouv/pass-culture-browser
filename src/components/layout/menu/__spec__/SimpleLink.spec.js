import { shallow } from 'enzyme'
import React from 'react'

import Item from '../Item'
import SimpleLink from '../SimpleLink'

describe('src | components | menu | SimpleLink', () => {
  let props

  beforeEach(() => {
    props = {
      item: {
        href: '/decouverte',
        icon: 'offres-w',
        target: '_blank',
        title: 'Decouverte',
      },
    }
  })

  it('should match the snapshot', () => {
    // given
    const wrapper = shallow(<SimpleLink {...props} />)

    // then
    expect(wrapper).toBeDefined()
    expect(wrapper).toMatchSnapshot()
  })

  describe('render()', () => {
    it('should render one link and one Item', () => {
      // given
      const wrapper = shallow(<SimpleLink {...props} />)

      // when
      const link = wrapper.find('a')
      const item = wrapper.find(Item)

      // then
      expect(link).toHaveLength(1)
      expect(item).toHaveLength(1)
    })

    it('should render a link with no rel if target equal blank', () => {
      // given
      props.item.target = 'plop'
      const wrapper = shallow(<SimpleLink {...props} />)

      // when
      const { rel } = wrapper.find('a').props()

      // then
      expect(rel).toBeNull()
    })
  })
})
