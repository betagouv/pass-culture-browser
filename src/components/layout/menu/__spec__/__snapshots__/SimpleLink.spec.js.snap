// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | menu | SimpleLink should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SimpleLink
    item={
      Object {
        "href": "/decouverte",
        "icon": "offres-w",
        "target": "_blank",
        "title": "Decouverte",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "/decouverte",
    "nodeType": "host",
    "props": Object {
      "children": <Item
        icon="offres-w"
        title="Decouverte"
      />,
      "className": "navlink mx12 flex-columns",
      "disabled": undefined,
      "href": "/decouverte",
      "rel": "noopener noreferer",
      "target": "_blank",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "icon": "offres-w",
        "title": "Decouverte",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "a",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "/decouverte",
      "nodeType": "host",
      "props": Object {
        "children": <Item
          icon="offres-w"
          title="Decouverte"
        />,
        "className": "navlink mx12 flex-columns",
        "disabled": undefined,
        "href": "/decouverte",
        "rel": "noopener noreferer",
        "target": "_blank",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "icon": "offres-w",
          "title": "Decouverte",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "a",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
